internal/nginx/config/validation/common.go

```diff
@@ -46,3 +47,25 @@ func validateEscapedStringNoVarExpansion(value string, examples []string) error
 	}
 	return nil
 }
+
+const invalidHostHeaderErrMsg string = "redefining the Host request header is not supported"
+
+// FIXME(ciarams87): Investigate validating NGINX specific header name rules here
```

Comment:
do we need to do any additional work?

Reply:
Addressed here https://github.com/nginxinc/nginx-kubernetes-gateway/pull/717/files#r1224687807